test/e2e/gpu_operator.go

```diff
@@ -0,0 +1,69 @@
+//go:build e2e
+// +build e2e
+
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	helmVals "helm.sh/helm/v3/pkg/cli/values"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/cluster-api/test/framework"
+)
+
+const (
+	nvidiaHelmChartRepoURL           string = "https://helm.ngc.nvidia.com/nvidia"
+	nvidiaGPUOperatorNamespace       string = "default"
+	nvidiaGPUOperatorHelmReleaseName string = "nvidia-gpu-operator"
+	nvidiaGPUOperatorHelmChartName   string = "gpu-operator"
+)
+
+// GPUOperatorSpecInput is the input for InstallGPUOperator.
+type GPUOperatorSpecInput struct {
+	BootstrapClusterProxy framework.ClusterProxy
+	Namespace             *corev1.Namespace
+	ClusterName           string
+}
+
+// InstallGPUOperator installs the official nvidia/gpu-operator helm chart.
+func InstallGPUOperator(ctx context.Context, inputGetter func() GPUOperatorSpecInput) {
+	var (
+		specName = "daemonsets"
+		input    GPUOperatorSpecInput
+	)
+
+	Expect(ctx).NotTo(BeNil(), "ctx is required for %s spec", specName)
+
+	input = inputGetter()
+	Expect(input.BootstrapClusterProxy).ToNot(BeNil(), "Invalid argument. input.BootstrapClusterProxy can't be nil when calling %s spec", specName)
+	Expect(input.Namespace).ToNot(BeNil(), "Invalid argument. input.Namespace can't be nil when calling %s spec", specName)
+	Expect(input.ClusterName).ToNot(BeEmpty(), "Invalid argument. input.ClusterName can't be empty when calling %s spec", specName)
+	clusterProxy := input.BootstrapClusterProxy.GetWorkloadCluster(ctx, input.Namespace.Name, input.ClusterName)
+	InstallNvidiaGPUOperatorChart(ctx, clusterProxy)
+}
+
+// InstallNvidiaGPUOperatorChart installs the official nvidia/gpu-operator helm chart
+// and validates that expected pods exist and are Ready.
+func InstallNvidiaGPUOperatorChart(ctx context.Context, clusterProxy framework.ClusterProxy) {
+	By("Installing nvidia/gpu-operator via helm")
+	values := &helmVals.Options{}
+	InstallHelmChart(ctx, clusterProxy, nvidiaGPUOperatorNamespace, nvidiaHelmChartRepoURL, nvidiaGPUOperatorHelmChartName, nvidiaGPUOperatorHelmReleaseName, values)
```

Comment:
I opted to not worry about identifying and waiting for specific daemonsets/pods like we do for cloud-provider-azure, as the specific deliverables of the nvidia gpu-operator helm chart aren't something we're able to predictively identify (IME) from version to version.

It seems like retry tolerance we have for identifying a `nvidia.com/gpu`-allocatable node is sufficient for this purpose, and that's the key outcome of the gpu-operator. And then of course we also run a CUDA test pod. (though I observed that it seems the gpu-operator helm chart actually does this itself now as a self-bootstrap test, so maybe we don't need to maintain that separate step anymore and we can trust that the application of the expected 'allocatable' metadata is a sufficiently strong signal of success?)

Reply:
>  validates that expected pods exist and are Ready.

Fair, let's remove this part from the comment?

Reply:
thx, fixed!